### ISSUE TITLE
Update Overlap Drop Parameter (small algorithmic update)

### ISSUE THREAD

**Issue by TomDonoghue** [voytekresearch/fooof#60]
I've been investigating the couple PSDs that fail fitting (it's not a big problem - literally a couple in towards a million PSDs in MEG dataset) - but it's slightly annoying in that they fail in very much edge case suboptimal guess gaussian parameters. In particular, I think it's when it tries to fit gaussians a bit close to the edge / each other. Small jitters can fix this (for example, moving f_range by 1 point). 

This got me playing a little with one of our 'magic' numbers - the degree of overlap with another gaussian at which we drop an oscillation guess before final fit (done in _drop_osc_overlap). Right now we have it set at 1 (and implicitly so, either way this should be noted as an internal parameter.)

There's no totally obvious way to choose this parameter in a purely data-driven approach (and it is currently 1.0 arbitrarily). Updating from 1.0 -> 1.5 or 2.0 (in units of bandwidth estimate of the gaussian guess) for example, marginally increases average error (on synthetic or real PSDs), but does reduce average number of gaussians fit (arguably reducing overfitting), so it's not entirely obvious what's better. I could do more explicit synthetic testing, but I reckon the answer becomes very dependent on the specific peak properties of the generated power spectra.

My quick impression from anecdotal fits, is that updating this number does reduce it being overzealous in uselessly adding more gaussians, so that's good (as this is one of our weakest points). It also does so in a manner not easily available to the user to control from public settings.

(Updating this upwards also seems to reduce likelihood of failing to fit, but from like 0.001% to something lower, so not really a relevant point). 

tl-dr: I think updating this number (increasing it from 1.0 to 1.5 or 2.0) is better, but exactly where to is unclear. My only hesitation is making a change to the algorithm so late (at worst, I think it'll be benign, but there is always a non-zero chance it has some negative effects somewhere). 

Thoughts?

Edit: I'm basically just thinking aloud here. I recommend updating this to 2.0.

**Comment by voytek** [comment]
Without sitting with the data more, it's hard for me to say. I just want to make sure you're not over-tuning to MEG. Have you tried performance on EEG and LFP to see what's better for those?

**Comment by TomDonoghue** [comment]
I scanned through the EEG & LFP data I have available as well. 

It's a relatively minor change, in that it has no effect whenever multiple gaussians are not near each other, so for many fits, this param is not invoked. 

Since we don't have an objective manner to set parameters like this, it's all a bit arbitrary, but my best-guess update is that it's better to push away from overlapping gaussians a bit more than we do currently (currently my impression is it ends up with unhelpful overlaps). 

So, I've updated to 1.5. 

We can revisit this (and other parameters) a bit later as we get more of a sense of how the algorithm is working across different datasets.
